### PR TITLE
Add seed params to search params

### PR DESCRIPTION
**Pull request by montanalow**
#164 & #166 didn't have a direct conflict, but they did race while adding both seeds and search params to certain classes. This completes them together.

### CODE REVIEW

**Comment by Mec-iS** [comment]
why tests did not spot these missing parts?

**Review by Mec-iS — APPROVED** [review]

**Comment by montanalow** [comment]
As far as I can tell, this code doesn't even compile. I think tests pass independently on branchs, and the breakage only shows up when both branches are merged into master.

File: src/cluster/kmeans.rs [hunk 1]
```diff
@@ -145,13 +145,17 @@ pub struct KMeansSearchParameters {
     pub k: Vec<usize>,
     /// Maximum number of iterations of the k-means algorithm for a single run.
     pub max_iter: Vec<usize>,
+    /// Determines random number generation for centroid initialization.
+    /// Use an int to make the randomness deterministic
+    pub seed: Vec<Option<u64>>,
```
Comment (morenol) [inline src/cluster/kmeans.rs:150]: Do we need Vec of Option? It looks to me that this should be either Option<u64> or Option<Vec<u64>>

Comment (montanalow) [reply src/cluster/kmeans.rs:150]: You're right, I think the cleanest type would be be Vec<u64>, although that'll require a little extra logic to convert to/from the Option<u64> seed type during iteration. I can take this up.

Comment (morenol) [reply src/cluster/kmeans.rs:150]: I am not completely sure if this is equivalent one-to-one, but in scikit it seems that random_state is only a number instead of a Vec, so not sure if we should try to match that here with a `Option<u64>` instead of the Vec

Comment (montanalow) [reply src/cluster/kmeans.rs:150]: I think you're looking at normal random_state params for estimators. Scikit grid search accepts a `Dict` with [values that are lists](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html#sklearn.model_selection.GridSearchCV). 

In theory, I agree that it would seem odd to me that people would want to grid search with multiple params, but in practice I think that making this the only *SearchParameter that is scalar would be more surprising.
